src/paste3/paste.py

```diff
@@ -917,31 +800,12 @@ def line_search_partial(
     cost_G : float
         The final cost after the update of the transport plan.
     """
-    combined_spatial_cost, a_gradient, b_gradient = ot.gromov.init_matrix(
```

Comment:
This function had its own implementation of computing gradient and cost, and was being passed as a parameter to the generic conditional gradient fn from ot, which then calls this function for line search. Now the cost fn and gradient fn are being passed as parameters.